sdk/tables/data-tables/src/serialization.ts

```diff
@@ -145,13 +145,51 @@ export function deserialize<T extends object = Record<string, any>>(
       if (`${key}@odata.type` in obj) {
         const type = (obj as any)[`${key}@odata.type`];
         typedValue = getTypedObject(value, type, disableTypeConversion);
+      } else if(["number", "string"].includes(typeof value)) {
+        // The service, doesn't return type metadata for number or strings
+        // if automatic type conversion is disabled we'll infer the EDM object
+        typedValue = inferTypedObject(key, value, disableTypeConversion);
```

Comment:
Seems like we could short-circuit this check sooner. 

I also wonder if it would be better to avoid the `[].includes` for comparing against two values since it feels like overkill, but I don't feel strongly about it.

```suggestion
      } else if(disableTypeConversion && ["number", "string"].includes(typeof value)) {
        // The service, doesn't return type metadata for number or strings
        // if automatic type conversion is disabled we'll infer the EDM object
        typedValue = inferTypedObject(key, value);
```